content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -1,8 +1,142 @@
+  dependencies</abbr> created with `dvc import-url`)
+- `repo`: This entry is only for external dependencies created with
+  `dvc import`, and can contains the following fields:
+
+  - `url`: URL of Git repository with source DVC project
+  - `rev`: Only present when the `--rev` option of `dvc import` is used.
+    Specific commit hash, branch or tag name, etc. (a
+    [Git revision](https://git-scm.com/docs/revisions)) used to import the
+    dependency from.
+  - `rev_lock`: Git commit hash of the external <abbr>DVC repository</abbr> at
+    the time of importing or updating (with `dvc update`) the dependency.
+
+Note that comments can be added to `.dvc` files and `dvc.yaml` using the
+`# comment` syntax. `meta` fields and `#` comments are preserved among
+executions of the `dvc repro` and `dvc commit` commands, but not when a `.dvc`
+file is overwritten by `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml files
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  features:
+    cmd: jupyter nbconvert --execute featurize.ipynb
+    deps:
+      - data/clean
+    params:
+      - levels.no
+    outs:
+      - features
+    metrics:
+      - performance.json
+  training:
+    cmd: python train.py
+    deps:
+      - train.py
+      - features
+    outs:
+      - model.pkl
+    plots:
+      - logs.csv:
+          x: epoch
+          x_label: Epoch
+    meta: 'For deployment'
+    # User metadata and comments are supported.
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`. Each stage can contain
+the possible following fields:
+
+- `cmd` (always present): Executable command defined in this stage
+- `deps`: List of <abbr>dependency</abbr> file or directory paths of this stage
+- `params`: List of the [parameters](/doc/command-reference/params). These are
+  key paths referring to another YAML file (`params.yaml` by default).
+- `outs`: List of <abbr>output</abbr> file or directory paths of this stage
+- `metrics`: List of [metric files](/doc/command-reference/metrics)
+- `plots`: List of [plot metrics](/doc/command-reference/plots) and optionally,
+  their default configuration (subfields matching the options of
+  `dvc plots modify`).
+- `frozen`: Whether or not this stage is frozen from reproduction
+- `always_changed`: Whether or not this stage is considered as changed by
+  commands such as `dvc status` and `dvc repro`. `false` by default
+- `meta` (optional): Arbitrary metadata can be added manually with this field.
+  Any YAML contents is supported. `meta` contents are ignored by DVC, but they
+  can be meaningful for user processes that read or write `.dvc` files directly.
+
+`dvc.yaml` files also support `# comments`. `meta` fields and `#` comments are
+always preserved in `dvc.yaml` stages.
```

Comment:
@jorgeorpinel we don't need to mention that they are preserved I think. That's what would 100% users expect as I mentioned before. It's confusing to see even this mentioned since you start thinking why is that important to mention, may something else is not preserved, etc.

Reply:
also, double check is they are preserved when we overwrite the stage

Reply:
I think we do because .dvc files has this note "not (preserverd) when a `.dvc` file is overwritten". But OK, removing from here. It's not a big deal either way.